### PR TITLE
fixing errors with regexfile and chmod/chown

### PR DESCRIPTION
**Pull request by AthrogatePEQ**
Fix erroneous chown/chmod messages during container startup.

## Description

The "prepare_configs()" subroutine in the bash_functions.sh file assumes that the regexFile variable will always be set and the file the variable points to will always exist.  I added conditional checks around the chown/chmod commands to only execute if the variable is set and the file exists.

## Motivation and Context

https://github.com/pi-hole/docker-pi-hole/issues/660

https://github.com/pi-hole/docker-pi-hole/issues/660

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: bash_functions.sh [hunk 1]
```diff
@@ -22,14 +22,20 @@ prepare_configs() {
...
     # not sure why pihole:pihole user/group write perms are not enough for web to write...dirty fix:
-    chmod 777 "${regexFile}"
+    if [[ -n ${regexFile} && -e ${regexFile} ]]; then 
+      chown pihole:pihole "${regexFile}"
+      chmod 777 "${regexFile}"
+    fi
     touch /var/log/pihole-FTL.log /run/pihole-FTL.pid /run/pihole-FTL.port /var/log/pihole.log
     chown pihole:pihole /var/run/pihole /var/log/pihole
     test -f /var/run/pihole/FTL.sock && rm /var/run/pihole/FTL.sock
     chown pihole:pihole /var/log/pihole-FTL.log /run/pihole-FTL.pid /run/pihole-FTL.port /etc/pihole /etc/pihole/dhcp.leases /var/log/pihole.log
```
Comment (casperklein) [inline bash_functions.sh:35]: Should `/etc/pihole/dhcp.leases` be removed from this line?

Comment (AthrogatePEQ) [reply bash_functions.sh]: good catch!

**Comment by AthrogatePEQ** [comment]
is anyone able to review this change so we could get it included on the next release, just minor startup error messages,  thanks

File: bash_functions.sh [hunk 2]
```diff
@@ -22,14 +22,20 @@ prepare_configs() {
     # Re-apply perms from basic-install over any volume mounts that may be present (or not)
     # Also  similar to preflights for FTL https://github.com/pi-hole/pi-hole/blob/master/advanced/Templates/pihole-FTL.service
     chown pihole:root /etc/lighttpd
-    chown pihole:pihole "${PI_HOLE_CONFIG_DIR}/pihole-FTL.conf" "/var/log/pihole" "${regexFile}"
+    chown pihole:pihole "${PI_HOLE_CONFIG_DIR}/pihole-FTL.conf" "/var/log/pihole" 
     chmod 644 "${PI_HOLE_CONFIG_DIR}/pihole-FTL.conf"
     # not sure why pihole:pihole user/group write perms are not enough for web to write...dirty fix:
-    chmod 777 "${regexFile}"
+    if [[ -n ${regexFile} && -e ${regexFile} ]]; then
```
Comment (dschaper) [inline bash_functions.sh:28]: Don't think we need both checks. `-e` should be enough, if the VAR is empty then the file won't exist.

File: bash_functions.sh [hunk 3]
```diff
@@ -22,14 +22,20 @@ prepare_configs() {
...
     # Also  similar to preflights for FTL https://github.com/pi-hole/pi-hole/blob/master/advanced/Templates/pihole-FTL.service
     chown pihole:root /etc/lighttpd
-    chown pihole:pihole "${PI_HOLE_CONFIG_DIR}/pihole-FTL.conf" "/var/log/pihole" "${regexFile}"
+    chown pihole:pihole "${PI_HOLE_CONFIG_DIR}/pihole-FTL.conf" "/var/log/pihole" 
     chmod 644 "${PI_HOLE_CONFIG_DIR}/pihole-FTL.conf"
     # not sure why pihole:pihole user/group write perms are not enough for web to write...dirty fix:
-    chmod 777 "${regexFile}"
+    if [[ -n ${regexFile} && -e ${regexFile} ]]; then 
+      chown pihole:pihole "${regexFile}"
+      chmod 777 "${regexFile}"
```
Comment (dschaper) [inline bash_functions.sh:30]: Can you verify that `rwx` is needed for this file still?

**Review by dschaper — CHANGES_REQUESTED** [review]

File: bash_functions.sh (hunk 3 above)
Comment (AthrogatePEQ) [reply bash_functions.sh:30]: I'm not sure i couldn't find anywhere in the script that the code was used.  i didn't want to remove it i just insulated it from spewing errors put in the logs.

Comment (gklanderman) [reply bash_functions.sh:30]: see my comment on Issue #660: that variable is never set so I would think the stray references should just be removed.

File: bash_functions.sh [hunk 4]
```diff
@@ -22,14 +22,20 @@ prepare_configs() {
...
+    fi
     touch /var/log/pihole-FTL.log /run/pihole-FTL.pid /run/pihole-FTL.port /var/log/pihole.log
     chown pihole:pihole /var/run/pihole /var/log/pihole
     test -f /var/run/pihole/FTL.sock && rm /var/run/pihole/FTL.sock
-    chown pihole:pihole /var/log/pihole-FTL.log /run/pihole-FTL.pid /run/pihole-FTL.port /etc/pihole /etc/pihole/dhcp.leases /var/log/pihole.log
+    chown pihole:pihole /var/log/pihole-FTL.log /run/pihole-FTL.pid /run/pihole-FTL.port /etc/pihole /var/log/pihole.log
+    if [[ -e /etc/pihole/dhcp.leases ]]; then 
+      chown pihole:pihole /etc/pihole/dhcp.leases 
+    fi 
     chmod 0644 /var/log/pihole-FTL.log /run/pihole-FTL.pid /run/pihole-FTL.port /var/log/pihole.log
```
Comment (gklanderman) [inline bash_functions.sh:39]: this seems like the right thing for the dhcp.leases warnings

**Comment by MichaIng** [comment]
I just ran across the same:
- `$regexFile` is set in `gravity.sh` which is never available at this point of the script: https://github.com/pi-hole/pi-hole/search?q=regexFile
  It should hence simply be removed.
- `/etc/pihole/dhcp.leases` is used by Pi-holes DHCP server? Since it is not always used, especially never present on first start, the file existence check looks good.
- 777 permissions should never be required. The comment talks about web access, so probably it was simply forgotten that the web interface runs as `www-data`?

Ah, if I see it right, regex.list is not used anymore but is now part of gravity.db: https://github.com/pi-hole/pi-hole/blob/master/gravity.sh#L247
That variable hence only exists for backwards-compatibility with Pi-hole pre-v5. I also verified that the web UI regex black/white lists work perfectly fine and regex.list gets never created:
```
2021-01-24 18:52:53 root@VM-Buster:/mnt/dietpi_userdata/docker-data/overlay2# docker exec -it pihole ls -l /etc/pihole/
total 4676
-rw-r--r-- 1 root   root        20 Jan 24 10:57 GitHubVersions
-rw-r--r-- 1 root   root        65 Jan 24 10:57 adlists.list
-rw-r--r-- 1 root   root         0 Jan 24 10:23 custom.list
-rw-r--r-- 1 root   root       618 Jan 24 11:42 dns-servers.conf
-rw-rw-r-- 1 pihole pihole 3493888 Jan 24 11:52 gravity.db
-rw-r--r-- 1 root   root   1200087 Jan 24 11:43 list.1.raw.githubusercontent.com.domains
-rw-r--r-- 1 root   root        36 Jan 24 11:43 local.list
-rw-r--r-- 1 root   root        20 Jan 24 11:50 localbranches
-rw-r--r-- 1 root   root        44 Jan 24 11:50 localversions
drwxr-xr-x 2 root   root      4096 Jan 24 10:23 migration_backup
-rw-r--r-- 1 pihole pihole       0 Jan 24 10:23 pihole-FTL.conf
-rw-r--r-- 1 root   root     53248 Jan 24 11:52 pihole-FTL.db
-rw-r--r-- 1 root   root       256 Jan 24 11:43 setupVars.conf
-rw-r--r-- 1 root   root       256 Jan 24 11:42 setupVars.conf.update.bak
```

**Comment by PromoFaux** [comment]
@AthrogatePEQ, please could you rebase your changes on the `dev` branch and make your PR against `dev` too? Thanks!

**Comment by PromoFaux** [comment]
Closing in favour of #787 (have cherry picked @AthrogatePEQ 's commits

**Pull request closed without merging**